adapters/flipp/flipp.go

```diff
@@ -123,7 +124,7 @@ func (a *adapter) processImp(request *openrtb2.BidRequest, imp openrtb2.Imp) (*a
 	var userKey string
 	if request.User != nil && request.User.ID != "" {
 		userKey = request.User.ID
-	} else if flippExtParams.UserKey != "" {
+	} else if flippExtParams.UserKey != "" && paramsUserKeyPermitted(request) {
 		userKey = flippExtParams.UserKey
```

Comment:
@hasan-kanjee should add/update json framework tests for new change

Reply:
added additional json framework tests